tests/js/server/aql/aql-profiler-noncluster.js

```diff
@@ -306,6 +308,90 @@ function ahuacatlProfilerTestSuite () {
       );
     },
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test EnumerateViewBlock1
+////////////////////////////////////////////////////////////////////////////////
+
+    testEnumerateViewBlock1: function () {
+      const col = db._create(colName);
+      const view = db._createView(viewName, "arangosearch", { links: { [colName]: { includeAllFields: true } } });
+      const prepare = (rows) => {
+        col.truncate();
+        col.insert(_.range(1, rows + 1).map((i) => ({value: i})));
+      };
+      const bind = () => ({'@view': viewName});
+      const query = `FOR d IN @@view SEARCH d.value != 0 OPTIONS { waitForSync: true } RETURN d.value`;
+
+      const genNodeList = (rows, batches) => {
+        return [
+          {type: SingletonBlock, calls: 2, items: 1},
+          {type: EnumerateViewNode, calls: batches + 1, items: rows},
```

Comment:
@KVS85 When `SingletonBlock` is called twice, something is wrong. On the first call, it should return a single line with state `DONE`. And no Block should call its parent again after it gets status `DONE`. Do you know perchance why this happens in the EnumerateViewBlock?

Also, these tests are for the blocks, not the nodes. So please always test the block that is actually instantiated, e.g. `IResearchViewOrderedBlock` or `IResearchViewUnorderedBlock` instead of `EnumerateViewNode`. Adapt `translateType` in `aql-profiler-test-helper.js` if needed to discern the two.

Reply:
@goedderz I know and https://github.com/arangodb/release-3.4/issues/134 was created to address `SingletonBock`. Once it's solved tests will start to fail and I will fix them. I don't know the root cause of this and, probably, it's just a matter of number in plan rather than actual double execution.

From `aql-profiler-test-helper.js` I see the following:
```
const IResearchViewUnorderedBlock = 'IResearchUnorderedViewNode';
const IResearchViewBlock = 'IResearchViewNode';
const IResearchViewOrderedBlock = 'IResearchOrderedViewNode';
```

Currently `IResearch*ViewNode` is not exposed anywhere even for **2**nd profiling level. There is always `EnumerateViewNode` only and no options to distinguish them (as advised to adapt `translateType`). Initially, 3 tests were made especially to use 3 different IResearch blocks.

Reply:
> Anyway, the type of block should be available to the explainer as well as for the tests.
`ExecutionNode` is meant to be treated as a factory for `ExecutionBlock` (with corresponding factory method `createBlock`), that's why block is implementation dependent and not visible to the end user. That's the crucial point of having factory there.
If one wants to check a type of the block returned by `ExecutionBlock::createBlock`, integration `js` tests is not a right place for that, `catch` tests must be used instead.

Reply:
> That's the crucial point of having factory there.
The point is that the actual types and how they're put together can be known at runtime only. That doesn't mean that the information isn't of interest.

Another point is that the tests here, and in several other places, are useless when they check the wrong implementation. I've actually found several issues related to that before this test suite was added. And as it is subject to change how a plan is instantiated, e.g. if optimizer rules are added, we should assert here that we do actually test what we intend to, otherwise code will silently switch to being untested later.

Reply:
@goedderz agree, and that's exactly for what `catch` tests were designed for. 
For `IResearchViewBlock`s we do exactly these kind of checks in catch tests, so no reason to have it publicly available in `js`, it just makes no sense.

Reply:
@goedderz e.g. `IResearchView::createBlock` may return different types of block based on the exact place where it resides

Reply:
I still think it's better to have the correctness of the tests to be as obvious as possible, which is not the case here. However, if you feel strongly against exposing this information, even in an undocumented profiling level, I'll still approve the PR.

But in that case, @KVS85, please add comments as to which block is tested in which test, and/or change the test names accordingly.

Reply:
@goedderz the reason I'm arguing against exposing such information to `js`, at least for iresearch blocks, is coming from extra burden to related to necessity of supporting artificial property and tightly coupling of tests and a code. Perhaps it makes sense (to some extent) for other blocks, because one may rely on a particular property, but for `IResearchBlock`s it doesn't work like this. That's why we're checking behavior in `catch` tests.

Reply:
@goedderz Let's merge it "as is" at the moment since `IResearch*ViewBlock` could be changed (or, at some point, event removed partially).

Reply:
@goedderz are you fine with merging it?

Reply:
yes.